File: backend/app/schemas/statistics.py
from pydantic import BaseModel
from typing import Optional


class DashboardStats(BaseModel):
    overall_accuracy: float
    total_questions: int
    completed_questions: int
    review_questions: int
    errors: int
    last_chapter_id: Optional[int] = None
    last_subject_id: Optional[int] = None
    last_chapter_name: Optional[str] = None


class ChapterStats(BaseModel):
    accuracy: float
    completed_questions: int
    remaining_questions: int
    review_count: int
    error_count: int
    almost_forgot_count: int

File: frontend/src/api/statistics.ts
import api from '@/lib/axios'

export interface DashboardStats {
  overall_accuracy: number
  total_questions: number
  completed_questions: number
  review_questions: number
  errors: number
  last_chapter_id?: number
  last_subject_id?: number
  last_chapter_name?: string
}

export interface ChapterStats {
  accuracy: number
  completed_questions: number
  remaining_questions: number
  review_count: number
  error_count: number
  almost_forgot_count: number
}

export const statisticsApi = {
  getDashboard: async (): Promise<DashboardStats> => {
    const response = await api.get('/statistics/dashboard')
    return response.data
  },
  getChapter: async (chapterId: number): Promise<ChapterStats> => {
    const response = await api.get(`/statistics/chapters/${chapterId}`)
    return response.data
  }
}
